MainaKamau92/apexselftaughtapi
# 166928070 Users should be able to create and update their profiles

## What does this PR do?
The PR creates mutations to allow users to create and update their profiles

## Description of task to be completed
Allow authenticated users to create and update their profiles
Inhibit alteration of profiles that do not belong to the logged-in user

#### How Has This Been Tested?
### Manually testing the PR
- Clone the repo https://github.com/MainaKamau92/apexselftaught.git
- Check out to ft-user-profile-166928070
- Run python manage.py makemigrations
- Run python manage.py migrate

### To create mutation use the below mutation
- Create User 
```
mutation {
  registerUser(email: "johndoe@googlemail.com", 
    mobileNumber: "0714136972", 
    password: "johndoe123", username: "johndoe") {
    user {
      id
    },
    errors
  }
}
```
- Login User after verifying your email
```
mutation {
  loginUser(email:"johndoe@googlemail.com", password:"johndoe123"){
    token
    error
  }
}
```
- Add the token generated to the header preceded by `JWT`
`JWT generated-token-here`
- Create  a profile
```
mutation {
  createProfile(
    avatar: "avatar.rr",
		country: "USA"
		county: "Boston"
		firstName: "John"
		github: "github/johndoe"
		industry: "Software"
		lastName: "Doe"
		secondaryEmail: ""
		userBio: "I love to coded"
		website: "johndoe.me"
  ){
    profile{
      id
    }
  }
}
```
- To update your profile 
```
mutation {
  updateProfile(
    id: 2,
    avatar: "avatar.rr",
		country: "Kenya"
		county: "Thika"
		firstName: "Lewis"
		github: "github/mainakamau92"
		industry: "Software"
		lastName: "Maina"
		secondaryEmail: "lewismaina@jack.com"
		userBio: "I love thinking"
		website: "lewismaina.me"
  ){
    message
    profile{
      id
      lastName
    }
  }
}
```
- Relevant PT Stories
- https://www.pivotaltracker.com/story/show/166928070
